### ISSUE TITLE
F-stack suitable for Ultra Low Latency http client?

### ISSUE THREAD

**Issue by moscovium115** [F-Stack/f-stack#818]
Hello,

I was wondering if F-stack would be suitable for a HTTP low latency client. I ask this since, there is only a server example present in the repo. (If I missed it, can someone link the HTTP client? )

Kind regards

**Comment by Frodocz** [comment]
Maybe check my naive implementation which might help: https://github.com/Frodocz/lepton

**Comment by jfb8856606** [comment]
Yes, F-Stack can be used as an HTTP client. It provides `ff_socket`, `ff_connect`, `ff_read`, `ff_write` APIs that support client-side TCP connections. The repo only ships server examples, but client implementations are straightforward using the ff_* API.

For a reference implementation, see the link shared above: https://github.com/Frodocz/lepton

For ultra-low latency, make sure to set in `config.ini`:
```ini
pkt_tx_delay=0
idle_sleep=0
net.inet.tcp.delayed_ack=0
```

Additionally, when running F-Stack as a multi-process client (multiple queues), source port selection involves `ff_rss_check` for RSS distribution. For low-latency scenarios, enable static `ff_rss_tbl` (for fixed 4-tuple connections) or `thash_adjust=1` (for dynamic 4-tuple connections) to optimize this path.

See also #798 for HTTPS client discussion. Closing as answered.
